runtime/api/Gamepads.py

```diff
@@ -51,18 +51,132 @@ def get_joysticks(index):
+    the y axis. -1 means the left joystick is pushed all the way forward.
+
+    :param index: The index of the gamepad, usually 0, 1, 2, or 3
+    :returns: A double between -1 and 1, negative is pushed forward.
+
+    :Examples:
+
+    >>> left_y = Gamepads.get_left_y(0)
+
+    """
+    return get_joysticks(index)[1]
+
+def get_right_x(index):
+    """Returns the position of the right joystick on the x axis.
+
+    The value is between -1 and 1, which represents where the right joystick is along
+    the x axis. -1 means the right joystick is pushed all the way to the left.
+
+    :param index: The index of the gamepad, usually 0, 1, 2, or 3
+    :returns: A double between -1 and 1, negative is pushed to the left.
+
+    :Examples:
+
+    >>> right_x = Gamepads.get_right_x(0)
+
+    """
+    return get_joysticks(index)[2]
+
+def get_right_y(index):
+    """Returns the position of the right joystick on the y axis.
+
+    The value is between -1 and 1, which represents where the right joystick is along
+    the x axis. -1 means the right joystick is pushed all the way forward.
+
+    :param index: The index of the gamepad, usually 0, 1, 2, or 3
+    :returns: A decimal between -1 and 1, negative is pushed to the forward
+
+    :Examples:
+
+    >>> right_y = Gamepads.get_right_y(0)
+
+    """
+    return get_joysticks(index)[3]
+
+def get_all_buttons(index):
     """Returns a list of button values corresponding to the specified gamepad.
 
-    Each button value is either a 0 (not pressed) or a 1 (pressed). Unlike
-    joysticks, there are no in-between values--it can only be a 0 or a 1. To
+    Each button value is either False (not pressed) or True (pressed). Unlike
+    joysticks, there are no in-between values--it can only be False or True. To
     see the exact mapping, click on the 'Details' button next to a gamepad in
     Dawn, or refer to https://w3c.github.io/gamepad/#remapping.
 
     :param index: The index of the gamepad, usually 0, 1, 2, or 3
-    :returns: A list of 0's and 1's, each corresponding to a button
+    :returns: A list of booleans, each corresponding to a button being pressed or not pressed
+
+    :Examples:
+
+    >>> all_buttons = Gamepads.get_all_buttons(0)
+    >>> all_buttons[1]
+    True
+    >>> all_buttons
+    [False, True, False, False, False, True, False, False, True, False, False, False]
+
+    """
+    return [x == 1 for x in mc.get('gamepad')[index]['buttons']]
+
+def get_button(index,button):
+    """Returns whether a button is pressed or not.
+
+    For a specific button (each button has has a number) the output is either 
+    True (pressed) or False (not pressed). To see the exact mapping, click on 
+    the 'Details' button next to a gamepad in Dawn, or refer to 
+    https://w3c.github.io/gamepad/#remapping.
+
+    :param index: The index of the gamepad, usually 0, 1, 2, or 3
+    :param button: The number of the button. See above for mappings.
```

Comment:
You should create an enum that maps from meaningful names (e.g. BUTTON_X, BUTTON_Y, and the like) and axis numbers